### PR TITLE
EMBCESSMOD-1289 - Supplier Portal Auth setup

### PR DESCRIPTION
**Pull request by josephweinkam**
Initial setup of keycloak authentication. Controllers all currently accept anonymous requests. Login component initiates login flow, and setup an auth guard for any routes under '/auth/'

### CODE REVIEW

**Comment by josephweinkam** [comment]
@ytqsl @SodhiA1

File: suppliers/src/API/EMBC.Suppliers.API/Startup.cs [hunk 1]
```diff
@@ -48,9 +55,63 @@ public Startup(IConfiguration configuration, IHostEnvironment env)
...
+                };
+                options.Validate();
+            });
+            services.AddAuthorization(options =>
+            {
+                options.AddPolicy(JwtBearerDefaults.AuthenticationScheme, policy =>
+                {
+                    policy.AddAuthenticationSchemes(JwtBearerDefaults.AuthenticationScheme)
+                        .RequireAuthenticatedUser()
+                        .RequireClaim("user_role")
```
Comment (ytqsl) [inline suppliers/src/API/EMBC.Suppliers.API/Startup.cs:105]: these are most probably not needed (yet)

File: suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs [hunk 2]
```diff
@@ -89,6 +92,22 @@ public ActionResult<Config> GetConfig()
             return Ok(result);
         }
 
+        [HttpGet("auth_config")]
+        [AllowAnonymous]
```
Comment (ytqsl) [inline suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Controllers/ConfigController.cs:96]: can this be merged with the other config endpoint?

File: suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Models/Data/countries.csv [hunk 3]
```diff
@@ -240,3 +240,4 @@
 "bdfbe6ef-ce96-ea11-b813-005056830319","ZMB","894","Zambia"
 "bffbe6ef-ce96-ea11-b813-005056830319","ZWE","716","Zimbabwe"
 "13539953-4195-ea11-b818-00505683fbf4","CAN","124","Canada"
+"8b500a47-2781-eb11-b825-00505683fbf4","W007","007","Wakanda"
```
Comment (ytqsl) [inline suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Models/Data/countries.csv:243]: ?

File: suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Models/Data/jurisdictions.csv [hunk 4]
```diff
@@ -160,3 +160,4 @@
 "ae6adfaf-9f97-ea11-b813-005056830319","Williams Lake","1","8262519f-0884-ea11-b813-005056830319"
 "b06adfaf-9f97-ea11-b813-005056830319","Zeballos","2","8262519f-0884-ea11-b813-005056830319"
 "1861651a-a097-ea11-b818-00505683fbf4","Northern Rockies","6","8262519f-0884-ea11-b813-005056830319"
+"81bf6c31-2781-eb11-b825-00505683fbf4","Gabriel's Jurisdiction","16","8262519f-0884-ea11-b813-005056830319"
```
Comment (ytqsl) [inline suppliers/src/API/EMBC.Suppliers.API/ConfigurationModule/Models/Data/jurisdictions.csv:163]: probably should be reverted

File: suppliers/src/API/EMBC.Suppliers.API/EMBC.Suppliers.API.csproj [hunk 5]
```diff
@@ -23,6 +23,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Jasper" Version="1.1.4" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="3.1.0" />
```
Comment (ytqsl) [inline suppliers/src/API/EMBC.Suppliers.API/EMBC.Suppliers.API.csproj:26]: Change to 3.1.* to ensure it picks up the lasted 3.1 package stream

File: suppliers/src/UI/embc-supplier/src/app/app.module.ts [hunk 6]
```diff
@@ -19,7 +20,13 @@ import { CoreModule } from './core/core.module';
     AppRoutingModule,
     LayoutModule,
     HttpClientModule,
-    ReactiveFormsModule
+    ReactiveFormsModule,
+    OAuthModule.forRoot({
+      resourceServer: {
+        customUrlValidation: url => url.startsWith('/api') && !url.endsWith('/configuration'),
+        sendAccessToken: true
```
Comment (ytqsl) [inline suppliers/src/UI/embc-supplier/src/app/app.module.ts:27]: This actually replaces the need to have the authguard, the library should send the token automatically

**Review by ytqsl — CHANGES_REQUESTED** [review]

**Review by SodhiA1 — APPROVED** [review]

**Review by ytqsl — APPROVED** [review]

**Pull request merged**